### ISSUE TITLE
data-id inside JSXExpressionContainer's value ....

### ISSUE THREAD

**Issue by mstruensee** [coderas/babel-plugin-jsx-remove-data-test-id#18]
Can the capability be added to remove data-id's from other attributes that are passed to other components as other props?

Something like this ... ?
```      
      it('removes data-test-id inside of other attributes that are passed in as objects', () => {
        const code = '<p someOtherAttribute={{ "data-test-id": "should-be-removed" }} someOtherAttributeKeep={{ className: "should-keep" }}></p>';
        const expectedCode = '<p someOtherAttributeKeep={{ className: "should-keep" }}></p>';
        const actual = x(code, { usePlugin: true });
        const expected = x(expectedCode);
        expect(uglify(actual)).to.equal(uglify(expected));
      });
```

I hacked something together [here](https://github.com/mstruensee/babel-plugin-jsx-remove-data-test-id/blob/master/src/index.js) ...

**Comment by coderas** [comment]
@mstruensee - firstly, sorry for the delay (once again I was in Greece and missed this). Secondly, I'd be really interested in the rationale behind why you'd need to do this. I'm exploring what `<p />` would be doing with that.

Thanks again for your time...

**Comment by mstruensee** [comment]
thanks for the response! I think it's hard to understand the reason when we look at a <p> tag ... think of it from a component perspective ... a team has a pattern library of UI components ... and we don't want to use data-test selectors but they don't put the dat-test where needed ... so we add something like ... <PLInput labelProps={ { "data-test": "my-test-selector" } } ... in this scenario the data-test is not removed because it's not an attribute to PLInput ... it's a key on an attribute ...

**Comment by coderas** [comment]
Ah I see, I hadn't thought about this.

If you do want to put an additive PR together, I think I could support that. Thanks for the insight.

**Comment by mstruensee** [comment]
sure, give me a bit ... in crunch time for black Friday

**Comment by coderas** [comment]
Closed by #23
